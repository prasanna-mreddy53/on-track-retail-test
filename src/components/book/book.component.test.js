import React from "react";
import Book from "./book.component";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

let book = {
  book_author: "test",
  book_title: "test book",
  book_publication_city: "london",
};

const wrapper = mount(<Book bookdet={book} />);

describe("Book Component Tests", () => {
  it("renders app correctly", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("should have the same props passed to the component", () => {
    expect(wrapper.props().bookdet).toEqual(book);
  });

  it("should match snapshot", () => {
    const tree = shallow(<Book bookdet={book} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
