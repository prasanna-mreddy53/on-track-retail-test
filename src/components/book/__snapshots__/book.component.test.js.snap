// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Component Tests should match snapshot 1`] = `
<Card
  body={false}
>
  <CardBody>
    <CardTitle>
      test
    </CardTitle>
    <CardSubtitle
      className="mb-2 text-muted"
    >
      test book
    </CardSubtitle>
    <CardText>
      london
    </CardText>
  </CardBody>
</Card>
`;
